backend/infrahub/computed_attribute/tasks.py

```diff
@@ -619,19 +608,53 @@ async def computed_attribute_setup_python(
                     identifier=computed_attribute.computed_attribute.key_name, scope=scope
                 )
                 await client.update_automation(automation_id=existing.id, automation=automation)
+                automations_to_keep.append(existing.id)
                 log.info(f"Query {computed_attribute.computed_attribute.key_name} Updated")
             else:
-                await client.create_automation(automation=automation)
+                automation_id = await client.create_automation(automation=automation)
+                automations_to_keep.append(automation_id)
                 log.info(f"Query {computed_attribute.computed_attribute.key_name} Created")
 
-            await service.workflow.submit_workflow(
-                workflow=TRIGGER_UPDATE_PYTHON_COMPUTED_ATTRIBUTES,
-                parameters={
-                    "branch_name": registry.default_branch,
-                    "computed_attribute_name": computed_attribute.computed_attribute.attribute.name,
-                    "computed_attribute_kind": computed_attribute.computed_attribute.kind,
-                },
-            )
+            if trigger_updates:
+                await service.workflow.submit_workflow(
+                    workflow=TRIGGER_UPDATE_PYTHON_COMPUTED_ATTRIBUTES,
+                    parameters={
+                        "branch_name": branch_name,
+                        "computed_attribute_name": computed_attribute.computed_attribute.attribute.name,
+                        "computed_attribute_kind": computed_attribute.computed_attribute.kind,
+                    },
+                )
+
+        automations_to_remove = existing_computed_attr_process_automations.return_obsolete(keep=automations_to_keep)
+        for automation_to_remove in automations_to_remove:
+            await client.delete_automation(automation_id=automation_to_remove)
+
+        automations_to_remove = existing_computed_attr_query_automations.return_obsolete(keep=automations_to_keep)
+        for automation_to_remove in automations_to_remove:
+            await client.delete_automation(automation_id=automation_to_remove)
```

Comment:
could the logic for handling query automation and process automation be refactored into a separate function or method that reduces duplication in this big function? maybe the answer is no

Reply:
Yes we should be able to have a catalogue of Automation with only the main information for each and reuse the code to configure prefect across most prefect automations

Reply:
I agree I think we need to come up with a format. Now we have a small helper class within the computed attribute domain to keep track of existing automations but I think it should be something more generic. I also don't like the calls we do for `async with get_client(sync_client=False) as client:` it should be something more generic that's perhaps found within `service.workflow` or similar. The setup of the AutomationCore is also very extensive where it could be 30 lines with a lot of boilerplate. I do however think that it will be a bit easier to see all this once we have a few of these functions and can decide what is required in an abstraction.

For now I've left it as is.